src/type-system/factories.ts

```diff
@@ -1,22 +1,22 @@
 import * as reflect from 'jsii-reflect';
 
-const STATIC_METHOD_SYM = Symbol();
-const CACHE: {
-  [STATIC_METHOD_SYM]?: any;
-} = {};
+const CACHE: Record<symbol, any> = {};
 
 /**
  * Return all static methods from this type system
  *
  * Cache on the type system to not have to do this calculation again and again.
  */
 export function allStaticMethods(ts: reflect.TypeSystem): reflect.Method[] {
-  if (CACHE[STATIC_METHOD_SYM]) {
-    return CACHE[STATIC_METHOD_SYM];
+  const cacheSymbol = Symbol(
+    ts.assemblies.map((a) => a.name + a.version).join()
+  );
+  if (CACHE[cacheSymbol]) {
+    return CACHE[cacheSymbol];
   }
 
   const methods = ts.classes.flatMap((x) => x.ownMethods);
-  return (CACHE[STATIC_METHOD_SYM] = methods);
+  return (CACHE[cacheSymbol] = methods);
```

Comment:
That's the badger were a cache was written that was later invalid for another test. With Jest we didn't have this issue, as every test file was a new process.

I've changed the cache symbol to use a string of all assembly names & versions. These should stay the same.

Note: In Lambda or local execution this won't be an issue since we use a new process every time. Only case would have been two `DeclarativeStacks` with different typesystems.